pkgs/development/tools/build-managers/bazel/bazel_7/default.nix

```diff
@@ -117,8 +101,161 @@ let
+    in
+    stdenv.mkDerivation {
+      name = "bazelDeps";
+      inherit src version;
+      sourceRoot = ".";
+      patches = [
+        # The repo rule that creates a manifest of the bazel source for testing
+        # the cli is not reproducible. This patch ensures that it is by sorting
+        # the results in the repo rule rather than the downstream genrule.
+        ./test_source_sort.patch
+      ];
+      patchFlags = [
+        "--no-backup-if-mismatch"
+        "-p1"
+      ];
+      nativeBuildInputs = [
+        unzip
+        runJdk
+        bazelForDeps
+      ] ++ lib.optional (stdenv.hostPlatform.isDarwin) libtool;
+      configurePhase = ''
+        runHook preConfigure
+
+        mkdir bazel_src
+        shopt -s dotglob extglob
+        mv !(bazel_src) bazel_src
+        mkdir vendor_dir
+
+        runHook postConfigure
+      '';
+      dontFixup = true;
+      buildPhase = ''
+        runHook preBuild
+        export HOME=$(mktemp -d)
+        (cd bazel_src; ${bazelForDeps}/bin/bazel --server_javabase=${runJdk} mod deps --curses=no;
+        ${bazelForDeps}/bin/bazel --server_javabase=${runJdk} vendor src:bazel_nojdk \
+        --curses=no \
+        --vendor_dir ../vendor_dir \
+        --verbose_failures \
+        --experimental_strict_java_deps=off \
+        --strict_proto_deps=off \
+        --tool_java_runtime_version=local_jdk_21 \
+        --java_runtime_version=local_jdk_21 \
+        --tool_java_language_version=21 \
+        --java_language_version=21)
+
+        # Some post-fetch fixup is necessary, because the deps come with some
+        # baggage that is not reproducible. Luckily, this baggage does not factor
+        # into the final product, so removing it is enough.
+
+        # the GOCACHE is poisonous!
+        rm -rf vendor_dir/gazelle~~non_module_deps~bazel_gazelle_go_repository_cache/gocache
+
+        # as is the go versions file (changes when new versions show up)
+        rm -f vendor_dir/rules_go~~go_sdk~go_default_sdk/versions.json
+
+        # and so are .pyc files
+        find vendor_dir -name "*.pyc" -type f -delete
+
+        # bazel-external is auto-generated and should be removed
+        # see https://bazel.build/external/vendor#vendor-symlinks for more details
+        rm vendor_dir/bazel-external
+
+        runHook postBuild
+      '';
+
+      installPhase = ''
+        mkdir -p $out/vendor_dir
+        cp -r --reflink=auto vendor_dir/* $out/vendor_dir
+      '';
+
+      outputHashMode = "recursive";
+      outputHash =
+        if stdenv.hostPlatform.system == "x86_64-linux" then
+          "sha256-II5R2YjaIejcO4Topdcz1H268eplYsYrW2oLJHKEkYw="
+        else if stdenv.hostPlatform.system == "aarch64-linux" then
+          "sha256-n8RMKf8OxJsEkcxLe7xZgMu9RyeU58NESFF9F0nLNC4="
+        else if stdenv.hostPlatform.system == "aarch64-darwin" then
+          "sha256-E6j31Sl+aGs6+Xdx+c0Xi6ryfYZ/ms5/HzIyc3QpMHY="
+        else
+          # x86_64-darwin
```

Comment:
technically it may cover other platforms too, like riscv or something

Reply:
That is not a [supported platform](https://nix.dev/manual/nix/2.17/installation/supported-platforms.html). I would be happy to take a vendor sha for i686, but since I think it is unlikely that people are waiting to be able to run Bazel >= 7.3 via Nix on a Raspberry Pi, I don't want to gate merging this PR, which has been outstanding for months, on its inclusion.

Reply:
What I meant is it's ok to only support most popular platforms, but there are more direct ways to express it:
- here instead of "else # assume x86_64-darwin" do "else if ...darwin case... else fail"
- and/or platforms meta attribute https://ryantm.github.io/nixpkgs/stdenv/meta/#var-meta-platforms

Reply:
Ah I understand, you're right, I can make that change.